P4-grpc-cve-process.md

```diff
@@ -0,0 +1,88 @@
+* Implemented in: N/A
+* Last updated: 2019-04-09
+* Discussion at: https://groups.google.com/d/msg/grpc-io/jhdCyvKfZlk/HUV675QwBQAJ
+## Abstract
+
+This gRFC proposes the CVE (Common Vulnerabilities and Exposure) process for gRPC. It is heavily adopted and influenced by similar processes that are in place for other Cloud Native Computing Foundation (CNCF) projects such as Kubernetes and Envoy.
+
+
+## Proposal
+
+This proposal defines the Common Vulnerabilities and Exposure (CVE) process for gRPC.  It is heavily adopted and influenced by similar processes that are in place for Kubernetes and Envoy.
+
+## Why do we need a CVE Process?
+
+gRPC is an RPC library with a 6 week release cycle that is in use by major companies, products and services today. The impact of a vulnerability can therefore have large economic, privacy and security implications.  
+
+A CVE disclosure process allows people to responsibly disclose these issues, and for the gRPC team to proactively address the issues and manage the communication around the disclosure. 
+
+
+## Disclosures
+
+### Private Disclosure Process
+
+The gRPC Community asks that all suspected vulnerabilities be privately and responsibly disclosed via the Private Disclosure process described below:
+
+To make a report, email the private grpc-security@googlegroups.com list with the security details and the [details expected for all gRPC bug reports](https://github.com/grpc/grpc/blob/master/.github/ISSUE_TEMPLATE.md).
+
+#### When should I report a vulnerability
+* You think you have discovered a potential security vulnerability in gRPC
+* You are unsure how a vulnerability affects gRPC
+* You think you discovered a vulnerability in another project that gRPC depends on.
+
+#### When should I not report a vulnerability
+* You need help tuning gRPC security
+* You need help applying security related updates
+* Your issue is not security related
+
+#### Security Vulnerability Response
+Each report is acknowledged and analyzed by Product Security Team (PST) members within 3 working days. 
+
+Any vulnerability information shared with PST stays within the gRPC project and will not be disseminated to other projects unless it is necessary to get the issue fixed.
+
+As the security issue moves from triage, to identified fix, to release planning we will keep the reporter updated.
+
+
+#### Public Disclosure Timing
+A public disclosure date is negotiated by the gRPC PST and the bug submitter. We prefer to fully disclose the bug as soon as possible once a user mitigation is available. It is reasonable to delay disclosure when the bug or the fix is not yet fully understood, the solution is not well-tested, or for vendor coordination. The timeframe for disclosure is from immediate (especially if it’s already publicly known) to a few weeks. As a basic default, we expect report date to disclosure date to be on the order of 7 days. The gRPC product security team holds the final say when setting a disclosure date.
```

Comment:
FWIW, we learned recently through Envoy's CVEs that 7 days from availability of candidate patches until public disclosure is likely too tight for most large organizations to work with if you are providing advanced notification.

Reply:
Based on your experience, should we relax this requirement to say.. order of 10 days?

Reply:
See https://github.com/envoyproxy/envoy/pull/6715 and discussion in https://github.com/envoyproxy/envoy/blob/master/security/postmortems/cve-2019-9900.md regarding this. Best practices seem to be 2-4 week window when you have distributors who need to patch based on a private fix release.

Reply:
changed from 1-7 to 14-20.. and it's loosely worded, so that it's not binding.